src/snowflake/snowpark/dataframe.py

```diff
@@ -3959,14 +3960,18 @@ def cache_result(
         if isinstance(self._session._conn, MockServerConnection):
             self.write.save_as_table(temp_table_name, create_temp_table=True)
         else:
-            create_temp_table = self._session._analyzer.plan_builder.create_temp_table(
-                temp_table_name,
-                self._plan,
-                use_scoped_temp_objects=self._session._use_scoped_temp_objects,
-                is_generated=True,
+            df = self._with_plan(
+                SnowflakeCreateTable(
+                    [temp_table_name],
+                    None,
```

Comment:
The SaveMode we passed here seems None, but when I look at the class init method for SnowflakeCreateTable, it seems it is not optional, which i think is correct, and we should have a save mode here. The SaveMode for SnowflakeCreateTable seems ERROR_IF_EXIST here, Let's make sure we have a SaveMode here. Also in the save_as_table function, we can add a check that only ERROR_IF_EXIST is supported for internally generated table. Then under  SaveMode == ERROR_IF_EXIST, you can add you if branch there to be clear